Formula/k0s.rb

```diff
@@ -0,0 +1,24 @@
+class K0s < Formula
+  desc "Zero Friction Kubernetes"
+  homepage "https://k0sproject.io"
+  url "https://github.com/k0sproject/k0s/archive/v0.7.0.tar.gz"
+  sha256 "5a697bf9f99c9dce7077b4bfa9323759cb54b49983b5544f1b1564447890dfef"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  uses_from_macos "curl"
+
+  def install
+    ENV.deparallelize
+    system "make",  "EMBEDDED_BINS_BUILDMODE=fetch", "VERSION=#{version}"
```

Comment:
Does this download dependencies?

Reply:
yes... its k0s thing that it bundles all needed kubernetes components.

Reply:
i think it doesnt... check the Makefile in `embedded-bins/`. it has fixed versions.

Reply:
a other alternative would be to download their already built file and use that.

Reply:
Fixed versions isn't the same as checksummed though. And downloading upstream's binaries wouldn't be acceptable either.

The only way this could be acceptable for homebrew is if it packaged, or even better, referenced the homebrew binaries.

Reply:
i will try to get my head around this... could you give me a hint why github actions complain that i installed a non executable? i tested it...works fine here on my box.

Reply:
Just tried it, and it isn't executable...on macOS.
```
$ file /usr/local/opt/k0s/bin/*
/usr/local/opt/k0s/bin/k0s: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=sdvMvbcL5Fkwss80veDu/dJI4um5KjOG9fnyvIBfw/NzaUE3_9qZHI8M20W6I_/x-1-5vlTdiBgmhlPB3TA, stripped
```
If it's meant to be Linux-only, the rule of thumb used to be "submit to https://github.com/Homebrew/linuxbrew-core", though since there are efforts to merge the two core repos, prepare to be rejected.

About the only way I think `k0s` will make it into Homebrew: convince the project maintainers to Distribute The Source and Nothing But The Source, instead of gluing together unauditable binary blobs.

Reply:
Since it's go there's nothing really preventing upstream from making it usable on macOS. This should be patched upstream instead of making it linux only.